api/hasura/actions/createEpoch.ts

```diff
@@ -0,0 +1,118 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { authCircleAdminMiddleware } from '../../../api-lib/circleAdmin';
+import { EPOCH_REPEAT } from '../../../api-lib/constants';
+import { formatShortDateTime } from '../../../api-lib/dateTimeHelpers';
+import { adminClient } from '../../../api-lib/gql/adminClient';
+import {
+  errorResponse,
+  errorResponseWithStatusCode,
+} from '../../../api-lib/HttpError';
+import {
+  createEpochInput,
+  composeHasuraActionRequestBody,
+} from '../../../src/lib/zod';
+
+async function handler(request: VercelRequest, response: VercelResponse) {
+  try {
+    const {
+      input: { payload: input },
+    } = composeHasuraActionRequestBody(createEpochInput).parse(request.body);
+    const { circle_id, repeat, start_date, days } = input;
+    let repeat_day_of_month = 0;
+    if (repeat > 0) {
+      const { epochs } = await adminClient.query({
+        epochs: [
+          {
+            limit: 1,
+            where: {
+              ended: { _eq: false },
+              circle_id: { _eq: circle_id },
+              repeat: { _gte: EPOCH_REPEAT.WEEKLY },
+            },
+          },
+          {
+            id: true,
+            start_date: true,
+            end_date: true,
+          },
+        ],
+      });
+      if (epochs.length) {
+        const overlappingEpoch = epochs.pop();
```

Comment:
This doesn't seem to check if it's actually overlapping, just if there's another repeating epoch set. Should this be named "repeatingEpoch" instead?

Reply:
updated this